doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst

```diff
@@ -0,0 +1,61 @@
+==================
+A four bar linkage
+==================
+
+The four bar linkage is a common example used in mechanics, which has to holonomic constraints. This example will make use of joints functionality provided in :mod:`sympy.physics.mechanics`. In summary we will use bodies and joints to define the open loop system. Next, we define the configuration constraints to close the loop. With the ``JointsMethod`` we get the input to manually setup the ``KanesMethod`` as its backend.
+
+.. image:: four_bar_linkage.svg
+   :align: center
+
+First we need to create the ``dynamicsymbols`` needed to describe the system as shown in the above diagram. In this case, the generalized coordinates `q_1`, `q_2` and `q_3` represent the angles between the links. Likewise, the generalized speeds `u_1`, `u_2` and `u_3` represent the angular velocities between the links. We also create some ``symbols`` to represent the lengths and density of the links. ::
+
+   >>> from sympy import symbols, Matrix, solve, simplify
+   >>> from sympy.physics.mechanics import *
+   >>> q1, q2, q3, u1, u2, u3 = dynamicsymbols('q1:4, u1:4')
+   >>> l1, l2, l3, l4, rho = symbols('l1:5, rho')
+
+With all symbols defined, we can now define the bodies. ::
+
+   >>> N = ReferenceFrame('N')
+   >>> inertias = [inertia(N, 0, 0, rho * l ** 3 / 12) for l in (l1, l2, l3, l4)]
+   >>> link1 = Body('Link1', frame=N, mass=rho * l1, central_inertia=inertias[0])
+   >>> link2 = Body('Link2', mass=rho * l2, central_inertia=inertias[1])
+   >>> link3 = Body('Link3', mass=rho * l3, central_inertia=inertias[2])
+   >>> link4 = Body('Link4', mass=rho * l4, central_inertia=inertias[3])
+
+Next, we also define the first three joints. ::
+
+   >>> joint1 = PinJoint('J1', link1, link2, coordinates=q1, speeds=u1,
+   ...                   parent_axis=link1.z, parent_joint_pos=l1 / 2 * link1.x,
+   ...                   child_axis=link2.z, child_joint_pos=-l2 / 2 * link2.x)
+   >>> joint2 = PinJoint('J2', link2, link3, coordinates=q2, speeds=u2,
+   ...                   parent_axis=link2.z, parent_joint_pos=l2 / 2 * link2.x,
+   ...                   child_axis=link3.z, child_joint_pos=-l3 / 2 * link3.x)
+   >>> joint3 = PinJoint('J3', link3, link4, coordinates=q3, speeds=u3,
+   ...                   parent_axis=link3.z, parent_joint_pos=l3 / 2 * link3.x,
+   ...                   child_axis=link4.z, child_joint_pos=-l4 / 2 * link4.x)
+
+Now we can formulate the holonomic constraint that will close the kinematic loop. ::
+
+   >>> loop = link4.masscenter.pos_from(link1.masscenter) \
+   ...        - l1 / 2 * link1.x + l4 / 2 * link4.x
+   >>> fh = Matrix([loop.dot(link1.x), loop.dot(link1.y)])
+
+In order to generate the equations of motions, we will use the ``JointsMethod`` as our fronted. Before setting up the ``KanesMethod`` as its backend we need to calculate the velocity constraints. ::
```

Comment:
Instead of "frontend" and "backend", maybe just say JointsMethod is used to book keep the open loop system.

Reply:
Yeah is probably clearer, used fronted and backend since [the other joints example](https://docs.sympy.org/latest/modules/physics/mechanics/examples/multi_degree_freedom_holonomic_system.html) also uses those terms.

Reply:
Ok, you can keep it then since we already did it. Just make sure the terms are defined somewhere if that is what we will use.